### PR TITLE
support limit/offset

### PR DESCRIPTION
**Pull request by danoctavian**
ticket 
https://qiibee.atlassian.net/browse/PD-123

i set the default limit to 100.

### CODE REVIEW

File: src/database/index.ts [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
   try {
-    // prevent SQL injections
-    const escapedAddress = conn.escape(address)
 
     const transactions = await conn.query(`SELECT * from transactions
     JOIN tokens ON transactions.contractAddress = tokens.contractAddress
-    WHERE toAddress=${escapedAddress} OR fromAddress=${escapedAddress}
-    ORDER BY blockNumber DESC`)
+    WHERE toAddress=? OR fromAddress=?
+    ORDER BY blockNumber DESC LIMIT ? OFFSET ?`, [address, address, limit, offset])
```
Comment (danoctavian) [inline src/database/index.ts]: this provides escaping from node-mysql, seems like the right way to always do it.

File: src/database/index.ts [hunk 2]
```diff
@@ -1,23 +1,27 @@
 import * as mysql from 'promise-mysql'
 import Config from './config'
+import { add } from "winston";
```
Comment (aaroncolaco) [inline src/database/index.ts]: Probably need to remove the semicolon to maintain a constant style of coding. Although I like semicolons ;)

Comment (danoctavian) [reply src/database/index.ts]: haha

File: __tests__/integration/transactions.test.ts [hunk 3]
```diff
@@ -198,6 +198,48 @@ describe('Transactions API Integration', () => {
...
+    // return value irrelevant for this test
+    ;(database.getTransactionHistory as any).mockImplementation(async () => [])
+
+    const transactionsAfterResponse = await request(app).get(`/transactions/${ACCOUNTS[0].address}/history`)
+
+    expect(transactionsAfterResponse.status).toBe(HttpStatus.OK)
+
+    expect(database.getTransactionHistory).toBeCalledWith(ACCOUNTS[0].address, 100, 0)
+  })
+
```
Comment (fedealconada) [inline __tests__/integration/transactions.test.ts]: should we add more tests? like for the edge cases? if you send negative limit and/or offset? of you send super big limit (but it should always be a max limit no matter if i enter 10000000000 as a limit), check if the offset is super big so its bigger the the amount of txs, what happens? etc

File: src/transactions/controller.ts [hunk 4]
```diff
@@ -66,11 +66,30 @@ const getTransaction = async (req, res) => {
   return res.json(tx) // TODO: improve response
 }
 
+const MAX_HISTORY_LIMIT = 100
```
Comment (fedealconada) [inline src/transactions/controller.ts]: +1

File: __tests__/integration/transactions.test.ts (hunk 3 above)
Comment (danoctavian) [reply __tests__/integration/transactions.test.ts]: sure

File: src/transactions/controller.ts (hunk 4 above)
Comment (cupaco) [reply src/transactions/controller.ts]: I think we should make MAX and DEFAULT_HISTORY_LIMIT separate.
